4_evaluation/src/modify_data.R

```diff
@@ -0,0 +1,33 @@
+################ function to read inds file and modify the data to remove all NA's
+
+modify_data <- function(in_ind){
+  dat_mod <- feather::read_feather(sc_retrieve(in_ind, remake_file = 'getters.yml')) %>%
+    pivot_longer(c(-seg_id_nat, -site_id, -date, -temp_c), names_to = 'model', values_to = 'predicted') %>%
+    filter(!is.na(predicted)) %>%
+    filter(!is.na(temp_c))
+
+  ## creating a dateframe (4 X 3) with the start and end date for each model .
+  dates <- group_by(dat_mod, model) %>%
+    summarize(start = min(date), end = max(date))
+  # finding the max/latest start and min/earliest date
+  start <- max(dates$start)
+  end <- min(dates$end)
+  ## filtering the date the and creating a subset with "rgcn_temp_c" model to select distinct seg_id_nat.
+  segs <- filter(dat_mod, model %in% 'rgcn_temp_c') %>%
+    select(seg_id_nat) %>% distinct() %>% pull(seg_id_nat)
+
```

Comment:
Did you verify in the metric csvs that the models have the same n value? e.g., that we filtered right and are comparing the same number of observations across models within any one of the csv files?